src/test/java/seedu/address/logic/commands/UndoCommandTest.java

```diff
@@ -0,0 +1,30 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalStudents.getTypicalTeachersPet;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+
+public class UndoCommandTest {
+    private Model model = new ModelManager(getTypicalTeachersPet(), new UserPrefs());
+
+    @Test
+    public void execute_success() {
+        UndoCommand undoCommand = new UndoCommand();
+        String expectedMessage = UndoCommand.MESSAGE_SUCCESS;
+        model.updateTeachersPetHistory();
+        model.updateTeachersPetHistory();
```

Comment:
I think it will be good to add comments explaining why there is 2 model.updateTeachersPetHistory(). Would be helpful for context!